src/ck-core/cksyncbarrier.C

```diff
@@ -43,8 +45,8 @@ LDBarrierClient CkSyncBarrier::AddClient(Chare* chare, std::function<void()> fn,
     // it can trigger. Do this asynchronously so that the caller functions for object
     // construction finish first.
     at_count += epoch - cur_epoch;
-    if (at_count >= clients.size())
-      thisProxy[thisIndex].CheckBarrier(false);
+    if (on && !startedAtSync && at_count >= clients.size() + 1)
```

Comment:
Can the inserting the LBClient object be moved above this code block to avoid the '+1' in the condition, so that the check is similar to the code in RemoveClient() fn definition.

Reply:
Done in 04825392ec79b38cccd8f65d55b89fce0b4a2a9c.